includes/class-wcpdf-main.php

```diff
@@ -128,6 +128,9 @@ public function attach_pdf_to_email ( $attachments, $email_id, $order, $email =
 			do_action( 'wpo_wcpdf_before_attachment_creation', $email_order, $email_id, $document_type );
 
 			try {
+				// log document generation to order notes
+				add_action( 'wpo_wcpdf_init_document', array( $this, 'log_email_attachment_to_order_notes' ) );
```

Comment:
Did you test if this results in duplicate notes when bulk completing (=bulk emailing?) orders? If so, we should do `remove_action` afterwards.

Reply:
Did not:

![Captura de ecrã de 2021-04-20 14-14-04](https://user-images.githubusercontent.com/533273/115401834-b9fd5400-a1e2-11eb-94b0-f6551e6f4b52.png)

Reply:
Is that the last order from the bulk series? (the first would only the function hooked once)

Reply:
I tested with 3 orders, all have one note.